server/src/main/java/org/elasticsearch/index/mapper/DocCountFieldMapper.java

```diff
@@ -96,17 +93,19 @@ protected void parseCreateField(ParseContext context) throws IOException {
         XContentParser parser = context.parser();
         XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, parser.currentToken(), parser);
 
-        long value = parser.longValue(false);
+        // Check that _doc_count is a single value and not an array
+        if (context.doc().getField(NAME) != null) {
```

Comment:
This method is a bit trappy as it does a linear scan of the fields of the document. Can you use `addWithKey` like `BinaryFieldMapper` does?